### PR TITLE
feat: add MigrateOrgCustomFileds interface

### PR DESCRIPTION
**Pull request by jing-zhi**
#### What this PR does / why we need it:
fix: fix p.BatchGetIssuePropertyInstances func
refactor: Unexposure OpenAPI and optimized code
fix: fix partial requirement list filtering does not include custom fields
refactor: use PropertyTypeEnum
feat: Change onlyProject to onlyCurrentScopeType
refactor: delete excess property-relation

#### Which issue(s) this PR fixes:
fix: fix p.BatchGetIssuePropertyInstances func
refactor: Unexposure OpenAPI and optimized code
fix: fix partial requirement list filtering does not include custom fields
refactor: use PropertyTypeEnum
feat: Change onlyProject to onlyCurrentScopeType
refactor: delete excess property-relation

- Fixes #your-issue_number
- [清理组织级别自定义事项](https://erda.cloud/erda/dop/projects/387/issues/all?id=567378&iterationID=12783&type=TASK)
- [协同事项筛选字段重复](https://erda.cloud/erda/dop/projects/387/issues/all?id=573504&iterationID=12783&type=BUG)
- [部分需求列表筛选不包含自定义字段](https://erda.cloud/erda/dop/projects/387/issues/all?id=574700&iterationID=12783&tab=ALL&type=BUG)

#### Specified Reviewers:

/assign @chengjoey @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       feat: add CleanOrgCustomFileds interface       |
| 🇨🇳 中文    |       添加迁移组织级字段接口，优化重名覆盖函数，修复筛选按钮重复       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/dop/providers/issue/dao/issue_property.go [hunk 1]
```diff
@@ -62,6 +65,100 @@ func (client *DBClient) UpdateIssueProperty(property *IssueProperty) error {
...
+		for _, guanxi := range xuYao {
+			client.Table("dice_issue_property_relation").Where("property_id = ?", guanxi.ID).Group("project_id").Find(&projectIDs)
+			// 创建
+			// 要怎么实现，只创建一次 common？？？？？？？？？？？？用 map 去重
+			for _, commonAndThree := range projectIDs {
+				// 创建个 map 不要重复了
+				var result IssueProperty
+				abc := guanxi.PropertyName + ":" + "project" + ":" + strconv.FormatInt(commonAndThree.PropertyID, 10)
+				fmt.Println(abc)
+				if v, ok := existCommon[guanxi.PropertyName+":"+"project"+":"+strconv.FormatInt(commonAndThree.ProjectID, 10)]; ok {
```
Comment (chengjoey) [inline internal/apps/dop/providers/issue/dao/issue_property.go:99]: implement a func to make project property key

Comment (jing-zhi) [reply internal/apps/dop/providers/issue/dao/issue_property.go]: okay, thank you for your suggestion. I tried a variable that might be easier

File: api/proto/dop/issue/core/core.proto [hunk 2]
```diff
@@ -200,6 +200,16 @@ service IssueCoreService {
     };
   };
 
+  rpc CleanOrgCustomFileds(CleanOrgCustomFiledsRequest)
```
Comment (sfwn) [inline api/proto/dop/issue/core/core.proto:203]: Use `Migrate` instead of `Clean`.

File: api/proto/dop/issue/core/core.proto [hunk 3]
```diff
@@ -200,6 +200,16 @@ service IssueCoreService {
...
+  returns (CleanOrgCustomFiledsResponse) {
+    option (google.api.http) = {
+      post: "/api/issues/actions/clear-org-customfileds",
+    };
+    option (erda.common.openapi) = {
```
Comment (sfwn) [inline api/proto/dop/issue/core/core.proto:208]: It's unnecessary to expose it as an OpenAPI.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/providers/issue/dao/issue_property.go [hunk 4]
```diff
@@ -62,6 +64,89 @@ func (client *DBClient) UpdateIssueProperty(property *IssueProperty) error {
...
+					Relation:          int64(resultCommon.ID), // 真的是从creatCommon可以拿到common的id吗？
+					Index:             commonProperty.Index,
+				}
+				var resultRelationshipType IssueProperty
+				client.Table("dice_issue_property").Create(&creatResultRelationshipType).Scan(&resultRelationshipType)
+				// 更新关系id
+				client.Table("dice_issue_property_relation").Where("org_id = ?", resultRelationshipType.OrgID).Where("project_id = ?", commonAndThree.ProjectID).
+					Where("property_id = ?", relationshipType.ID).Updates(IssuePropertyRelation{PropertyID: int64(resultRelationshipType.ID)})
+				// 因为 value 连接的 property_id，是连接的组织级到项目级别下，有新的 property_id,所以需要新创建value
+				if commonProperty.PropertyType == "Select" || commonProperty.PropertyType == "MultiSelect" {
```
Comment (sfwn) [inline internal/apps/dop/providers/issue/dao/issue_property.go:128]: use `PropertyTypeEnum`.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/dop/issue/core/core.proto (hunk 2 above)
Comment (jing-zhi) [reply api/proto/dop/issue/core/core.proto:203]: ok,thanks

File: api/proto/dop/issue/core/core.proto [hunk 5]
```diff
@@ -462,6 +472,16 @@ message GetIssuePropertyInstanceResponse {
   IssueAndPropertyAndValue data = 1;
 }
 
+message CleanOrgCustomFiledsRequest {
+  int64 orgID = 1;
+  repeated string propertyName = 2;
```
[the remark this answers is not in the document]
Comment (jing-zhi) [reply api/proto/dop/issue/core/core.proto:477]: ok,thanks.

File: internal/apps/dop/providers/issue/dao/issue_property.go [hunk 6]
```diff
@@ -62,6 +64,89 @@ func (client *DBClient) UpdateIssueProperty(property *IssueProperty) error {
...
+					for _, v := range beforePropertyValue {
+						creatPropertyValue := IssuePropertyValue{
+							PropertyID: int64(resultCommon.ID),
+							Value:      v.Value,
+							Name:       v.Name,
+						}
+						// 创建一个新的，修改一个 relation 关系
+						var currentValue IssuePropertyValue
+						client.Table("dice_issue_property_value").Create(&creatPropertyValue).Scan(&currentValue)
+						// 找到之前调用的老的，找到的话就更新为新的value_id，找不到就算了
```
[the remark this answers is not in the document]
Comment (jing-zhi) [reply internal/apps/dop/providers/issue/dao/issue_property.go:141]: ok,thanks.

File: api/proto/dop/issue/core/core.proto (hunk 3 above)
Comment (jing-zhi) [reply api/proto/dop/issue/core/core.proto:208]: ok

File: api/proto/dop/issue/core/core.proto [hunk 7]
```diff
@@ -462,6 +472,16 @@ message GetIssuePropertyInstanceResponse {
...
+}
+
+message CleanOrgCustomFiledsResponse {
+  string ok = 1;
```
[the remark this answers is not in the document]
Comment (jing-zhi) [reply api/proto/dop/issue/core/core.proto:481]: ok

File: internal/apps/dop/providers/issue/dao/issue_property.go [hunk 8]
```diff
@@ -62,6 +64,89 @@ func (client *DBClient) UpdateIssueProperty(property *IssueProperty) error {
...
+				creatResultRelationshipType := IssueProperty{
+					ScopeType:         "project",
+					ScopeID:           commonAndThree.ProjectID,
+					OrgID:             commonProperty.OrgID,
+					Required:          commonProperty.Required,
+					PropertyType:      commonProperty.PropertyType,
+					PropertyName:      commonProperty.PropertyName,
+					DisplayName:       commonProperty.DisplayName,
+					PropertyIssueType: relationshipType.PropertyIssueType,
+					Relation:          int64(resultCommon.ID), // 真的是从creatCommon可以拿到common的id吗？
```
[the remark this answers is not in the document]
Comment (jing-zhi) [reply internal/apps/dop/providers/issue/dao/issue_property.go:119]: ok

File: internal/apps/dop/providers/issue/dao/issue_property.go (hunk 4 above)
Comment (jing-zhi) [reply internal/apps/dop/providers/issue/dao/issue_property.go]: ok

**Review by sfwn — APPROVED** [review]

**Pull request merged**